si_bandit/cli.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+# si-bandit
+# Copyright(C) 2020 Kevin Postlethwait.
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""This is the main script of the template project."""
+from version import __version__ as si_bandit_version
+from version import __title__ as si_bandit_title
+import os
+import subprocess
+import tempfile
+import tarfile
+import click
+import json
+from typing import Optional
+from bandit import __version__ as bandit_version
+from pprint import PrettyPrinter
+import datetime
+
+from thoth.analyzer import run_command
+from thoth.analyzer import print_command_result
+from thoth.common import init_logging
+
+init_logging()
+
+
+@click.command()
+@click.pass_context
+@click.option(
+    "--output",
+    "-o",
+    type=str,
+    default="-",
+    envvar="THOTH_SI_BANDIT_OUTPUT",
+    help="Output file to print results to.",
+)
+@click.option(
+    "--from-directory",
+    "-d",
+    type=str,
+    envvar="THOTH_SI_BANDIT_DIR",
+    help="Input directory for running bandit.",
+)
+@click.option(
+    "--package-name",
+    "-n",
+    required=True,
+    type=str,
+    envvar="THOTH_SI_BANDIT_PACKAGE_NAME",
+    help="Name of package bandit is being run on.",
+)
+@click.option(
+    "--package-version",
+    type=str,
+    required=True,
+    envvar="THOTH_SI_BANDIT_PACKAGE_VERSION",
+    help="Version to be evaluated.",
+)
+@click.option(
+    "--package-index",
+    type=str,
+    default="https://pypi.org/simple",
+    envvar="THOTH_SI_BANDIT_PACKAGE_INDEX",
+    help="Which index is used to find package.",
+)
+@click.option("--no-pretty", is_flag=True, help="Do not print results nicely.")
+def si_bandit(
+    click_ctx,
+    output: Optional[str],
+    from_directory: Optional[str],
+    package_name: str,
+    package_version: Optional[str],
+    package_index: Optional[str],
+    no_pretty: bool,
+):
+    """Run the cli for si-bandit."""
+    if from_directory is None:
+        d = tempfile.TemporaryDirectory()
+        command = (
+            f"pip download --no-binary=:all: --no-deps -d {d.name} -i {package_index} "
+            + f"{package_name}=={package_version}"""
```

Comment:
Use [arbitrary equality](https://www.python.org/dev/peps/pep-0440/#arbitrary-equality) here:

```
            f"{package_name}==={package_version}"""
```

Otherwise `1.0` will be equal to `1.0.post0` and that's what we do not want.

BTW strings can be escaped to avoid instructions needed for concatenating strings.